### PR TITLE
Features/popup

### PR DESCRIPTION
**Pull request by shirrotem**

### CODE REVIEW

File: src/data/types/cardProps.ts [hunk 1]
```diff
@@ -6,6 +6,6 @@ interface CardProps{
     icon?: string;
     price?: string;
     type: string;
-
+    toggleModal? : (dishName: string)=>void;
```
Comment (MoranOshiaMoveo) [inline src/data/types/cardProps.ts]: change it to more intuitive name like handleOnClick

File: src/shared/components/Modal/ModalPortal.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+const ModalPortal: React.FC<ModalPortalProps> = ({ children }) => {
+  const modalRoot = document.getElementById('modal-root');
+  if (!modalRoot) return null;
+
+  return ReactDOM.createPortal(children, modalRoot);
+};
+
+export default ModalPortal;
```
Comment (MoranOshiaMoveo) [inline src/shared/components/Modal/ModalPortal.tsx]: irrelevant file, you don't need it

File: src/data/types/carouselProps.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ interface CarouselProps {
     type: any;
     typeName: string;
     title: string;
+    toggleModal? : (dishName: string)=>void;
```
Comment (MoranOshiaMoveo) [inline src/data/types/carouselProps.ts]: same

File: src/pages/home-page/components/HomePage/HomePage.tsx [hunk 4]
```diff
@@ -7,16 +7,42 @@ import './HomePage.scss';
...
+import Modal from "../../../../shared/components/Modal/Modal";
+import ModalPortal from "../../../../shared/components/Modal/ModalPortal";
+import Card from "../../../../shared/components/Card/Card";
+import { Dish } from "../../../../data/types/data";
 
 const HomePage = () => {
+    const [dishForModal, setDishForModal] = useState<Dish | null>(null);
+    const [showModal, setShowModal] = useState(false);
+
+    const toggleModal = (dishName: string = '') => {
```
Comment (MoranOshiaMoveo) [inline src/pages/home-page/components/HomePage/HomePage.tsx]: same change to handleOnClick for exmple, and use useCallBack

File: src/shared/components/Modal/Modal.scss [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    align-items: center;
+    z-index: 2;
+  }
+  
+  .modal-content {
+    display: flex;
+    padding-bottom: 0px;
+    flex-direction: column;
+    align-items: center;
+    gap: 40px;
```
Comment (MoranOshiaMoveo) [inline src/shared/components/Modal/Modal.scss]: check if relevant

File: src/shared/components/Modal/Modal.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+interface ModalProps {
+  onClose: () => void;
+  children: React.ReactNode;
+}
+
+const Modal: React.FC<ModalProps> = ({ onClose, children }) => {
+  return (
+    <div className="modal-overlay" onClick={onClose}>
+      <div onClick={(e) => e.stopPropagation()}>
```
Comment (MoranOshiaMoveo) [inline src/shared/components/Modal/Modal.tsx]: not need and irrelevant

**Review by MoranOshiaMoveo — CHANGES_REQUESTED** [review]

File: src/data/types/cardProps.ts (hunk 1 above)
Comment (shirrotem) [reply src/data/types/cardProps.ts]: fixed

File: src/data/types/carouselProps.ts (hunk 3 above)
Comment (shirrotem) [reply src/data/types/carouselProps.ts]: fixed

File: src/shared/components/Modal/ModalPortal.tsx (hunk 2 above)
Comment (shirrotem) [reply src/shared/components/Modal/ModalPortal.tsx]: fixed

File: src/pages/home-page/components/HomePage/HomePage.tsx (hunk 4 above)
Comment (shirrotem) [reply src/pages/home-page/components/HomePage/HomePage.tsx]: fixed

File: src/shared/components/Modal/Modal.scss (hunk 5 above)
Comment (shirrotem) [reply src/shared/components/Modal/Modal.scss]: fixed

File: src/shared/components/Modal/Modal.tsx (hunk 6 above)
Comment (shirrotem) [reply src/shared/components/Modal/Modal.tsx]: fixed

**Review by MoranOshiaMoveo — APPROVED** [review]

**Pull request merged**